### PR TITLE
Renamed warm strategy opts

### PR DESCRIPTION
**Pull request by mpirvu**
The default optimization strategy for the warm optimization level is `cheapWarmStrategyOpts`, while the older, `warmStrategyOpts` is not used (unless a specific option is given). This commit renames
warmStrategyOpts to oldWarmStrategyOpts
and
cheapWarmStrategyOpts to warmStrategyOpts

### CODE REVIEW

**Comment by mpirvu** [comment]
@vijaysun-omr Could you please review this PR? Thank you

**Review by vijaysun-omr — APPROVED** [review]

**Comment by vijaysun-omr** [comment]
Jenkins test sanity all jdk11

**Comment by mpirvu** [comment]
Tests on mac failed due to infra:
```
19:42:07  Exception: org.jenkinsci.plugins.workflow.support.steps.AgentOfflineException: Unable to create live FilePath for mac10-x86-3; mac10-x86-3 was marked offline: Connection was broken
[Pipeline] End of Pipeline
Finished: FAILURE
```
On Power there were CRIU failures, but not caused by this PR.

**Comment by vijaysun-omr** [comment]
Checks have passed except for the known issues mentioned. This is a cosmetic change anyway to make it easier to read the code and reason about the warm strategy in use. Merging.

File: runtime/compiler/optimizer/J9Optimizer.cpp [hunk 1]
```diff
@@ -1027,14 +1027,7 @@ J9::Optimizer::optimizationStrategy(TR::Compilation *c)
...
    TR_Hotness strategy = c->getMethodHotness();
-   if (strategy == warm && !c->getOption(TR_DisableCheapWarmOpts))
-      {
-      return cheapWarmStrategyOpts;
-      }
-   else
-      {
-      return j9CompilationStrategies[strategy];
-      }
+   return j9CompilationStrategies[strategy];
```
Comment (vijaysun-omr) [inline runtime/compiler/optimizer/J9Optimizer.cpp:1030]: Is there a cleanup task (separate PR) to free up the option bit for `TR_DisableCheapWarmOpts` ?

**Pull request merged**